### PR TITLE
updated cart page and functionality with APIs

### PR DESCRIPTION
**Pull request by manojsarna**
### Features Added 

**- [x] Pages - CartPage**
**- [x] Functionality - Integrated - Add To Cart, Remove From Cart , Update Cart Quantity, Add to wishlist**
**- [x] Features - Mobile Responsive* *
**- [x] Features - You can go to Product Page by clicking On Image or Title of the Product Card* *

Deploy Link to Check : https://hopeful-newton-961563.netlify.app/

### Files To Check
[/cart-context.js](https://github.com/manojsarna/manoj-smash-cart-ecommerce-react/compare/cartPage-API-functionality?expand=1#diff-b372b538d824a4879ea6eaf5c1879a1fe74a5128f71b912e04be9a462b2ac0e4)
[Cart.jsx](https://github.com/manojsarna/manoj-smash-cart-ecommerce-react/compare/cartPage-API-functionality?expand=1#diff-311e158bc248b4a335dc79bc4d3d5e9fb0f3ef7d988abbdfea29c6e1614d2833)
### Ignore Rest of the Files
image
image

### CODE REVIEW

**Review by b0llu — APPROVED** [review]
Short PR. Clean code everything is looking good. You can merge this PR

File: src/pages/cart/Cart.jsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      cartItems: acc.cartItems + currentProduct.qty,
+      totalPrice:
+        acc.totalPrice + currentProduct.price.current * currentProduct.qty,
+    }),
+    initialCheckoutState
+  );
+  return (
+    <main className="sm-main-cart">
+      <div className="sm-cart-products-list">
+        {cart.length === 0 && (
```
Comment (dev-ashishsethi) [inline src/pages/cart/Cart.jsx:34]: instead of using shortcircuiting you can use ternary operator

File: src/pages/cart/cart.css [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+  width: 24rem;
+  cursor: default;
+  flex-direction: row;
+  justify-content: flex-start;
+  height: 20rem;
+  overflow: hidden;
+  min-width: 100%;
+}
+
+.sm-cart-products-list .sm-card .sm-card-img {
```
Comment (PuneetP16) [inline src/pages/cart/cart.css:45]: Suggestion: you could group similar css properties to avoid repetitioin.

File: src/pages/cart/cart.css [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+.sm-card-hor .sm-card-content .sm-card-content-brand .sm-card-rating {
+  margin: 0;
+}
+.sm-card-content-brand-new {
+  font-size: 1.3rem;
+  display: flex;
+  align-items: center;
+  gap: 4rem;
+  /* justify-content: space-between; */
```
Comment (PuneetP16) [inline src/pages/cart/cart.css:179]: remove commented code

**Review by PuneetP16 — APPROVED** [review]
Looks good to merge. API functionality is working properly

**Review by dev-ashishsethi — APPROVED** [review]
good to see that you are abstracting your axios calls and other functionalities

**Review by iRohitGaur — APPROVED** [review]
Nice work! Good to merge 👍🏻

**Review by sudipkundu999 — APPROVED** [review]
Good work. Good to merge

**Review by pankaj-prog — APPROVED** [review]
You could move cart util functions to a different file, It will make your code more readable.
Other than this the code is readable and easily understandable

File: src/context/cart-context/cart-context.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+      });
+      if (response.status === 200) {
+        setCart(response.data.cart);
+        dispatch({
+          type: "TOAST_SUCCESS",
+          payload: "Removed from Cart",
+        });
+      }
+    } catch (error) {
+      console.error(error.response.data.errors);
```
Comment (rahulyadav139) [inline src/context/cart-context/cart-context.js:75]: Suggestion: Instead of console.log You can use toast to show an error to the user.

**Review by rahulyadav139 — APPROVED** [review]

**Pull request merged**